### PR TITLE
Add retry mechanism for the DB connection

### PR DESCRIPTION
**Pull request by stin911**
This solves #1479

### CODE REVIEW

File: vantage6-backend-common/vantage6/backend/common/base.py [hunk 1]
```diff
@@ -99,31 +103,46 @@ def _connect(
...
+            except OperationalError as e:
+                log.error(f"Connection attempt failed: {str(e)}")
+
+                # Check if the maximum retry duration has been exceeded
+                if attempt < MAX_ATTEMPT - 1:
+                    log.info(f"Retrying in {RETRY_DELAY} seconds...")
+                    sleep(RETRY_DELAY)
+                else:
+                    raise Exception(
+                        "Unable to connect to the database after 5 minutes of attempts. Please ensure the database is up and running.") from e
```
Comment (tim-hendriks) [inline vantage6-backend-common/vantage6/backend/common/base.py]: The exception message here is hardcoded to 5 minutes, but I suppose that duration depends on the variables set earlier

File: vantage6-backend-common/vantage6/backend/common/base.py [hunk 2]
```diff
@@ -87,6 +88,9 @@ def _connect(
         """
         self.allow_drop_all = allow_drop_all
         self.URI = uri
+        # Number of attempts and delay in second for the db connection
```
Comment (tim-hendriks) [inline vantage6-backend-common/vantage6/backend/common/base.py:91]: Should be fine like this, but if this were Spring Boot I would make these variables optionally configurable from the config file. Not sure if it's as straightforward to do that in this case. 
For readability of the code I prefer things like MAX_NUMBER_OF_ATTEMPTS, RETRY_DELAY_IN_SECONDS as opposed to adding comments.

**Review by tim-hendriks — APPROVED** [review]
I added two very nitpicky comments more from an exchanging ideas standpoint. Feel free to ignore them.

File: vantage6-backend-common/vantage6/backend/common/base.py (hunk 2 above)
Comment (stin911) [reply vantage6-backend-common/vantage6/backend/common/base.py:91]: Yeah I thought the same.

File: vantage6-backend-common/vantage6/backend/common/base.py [hunk 3]
```diff
@@ -99,31 +103,49 @@ def _connect(
...
+            except OperationalError as e:
+                log.error(f"Connection attempt failed: {str(e)}")
+
+                # Check if the maximum retry duration has been exceeded
+                if attempt < MAX_ATTEMPT - 1:
+                    log.info(f"Retrying in {RETRY_DELAY} seconds...")
+                    sleep(RETRY_DELAY)
+                else:
+                    raise Exception(
+                        "Unable to connect to the database after 5 minutes of attempts. Please ensure the database is up and running."
```
[the remark this answers is not in the document]
Comment (stin911) [reply vantage6-backend-common/vantage6/backend/common/base.py]: I changed this in the last two commits.

File: vantage6-backend-common/vantage6/backend/common/base.py [hunk 4]
```diff
@@ -87,6 +88,9 @@ def _connect(
         """
         self.allow_drop_all = allow_drop_all
         self.URI = uri
+        # Number of attempts and delay in second for the db connection
+        MAX_ATTEMPT = 10
+        RETRY_DELAY = 30  # seconds
```
[the remark this answers is not in the document]
Comment (stin911) [reply vantage6-backend-common/vantage6/backend/common/base.py]: yeah maybe is the best approach if they are going to set them up somewhere. 
\

[the remark this answers is not in the document]
Comment (stin911) [reply vantage6-backend-common/vantage6/backend/common/base.py]: I will make another commit

**Pull request merged**